### ISSUE TITLE
consider adding typescript declaration file

### ISSUE THREAD

**Issue by DavidBabel** [bkrem/react-d3-tree#171]
Here is what i did in my project, it's not complete but it fit my needs and is a good basis to implement it : 

```js
declare module 'react-d3-tree' {
  type ReactD3TreeItem = {
    name: string;
    attributes?: {
      [key: string]: string;
    };
    children?: ReactD3TreeItem[];
  };

  type ReactD3TreeProps = {
    data: ReactD3TreeItem[];
    orientation?: 'horizontal' | 'vertical';
    collapsible?: boolean;
    separation?: { siblings: number; nonSiblings: number };
    nodeSize?: { x: number; y: number };
    zoom?: number;
    scaleExtent?: { min: number; max: number };
  };

  var Tree: React.ComponentClass<ReactD3TreeProps, any>;
  export default Tree;
}

```

**Comment by bkrem** [comment]
Thanks for this @DavidBabel! I'll complete the type definitions for the remaining props and get it published. 
I think you should be properly credited for this as a contributor, so feel free to open a PR with your type definitions if you like.

This is also a great starting point for a transition to Typescript, which I want to get done once v2 is finally published (hopefully by the end of the holiday season).

**Comment by DavidBabel** [comment]
No problem, don't worry for the contrib, not a big deal to me.
Glad to help anyway

**Comment by bkrem** [comment]
Released as [v1.12.2](https://github.com/bkrem/react-d3-tree/releases/tag/v1.12.2), closing.
